### PR TITLE
refactored hasStore boolean into hasStore() function

### LINKED ISSUE #70: Refactor hasStore so that it is a function and not a boolean
### Mission Control

The DAV project you are looking at is *Mission Control*. It is the brain in charge of orchestrating missions between DAV users and autonomous vehicles.

### The Issue

Within `/server/lib/environment.js` is a boolean called `hasStore` which contains true or false, based on the availability of Redis in the current environment. It is currently a boolean, but it should be a function that returns a boolean instead.

In other words:

```javascript
const { hasStore } = require('./server/lib/environment.js');

// This is how we use it now:
if ( hasStore )   useRedis();

// This is how we want to use it:
if ( hasStore() ) useRedis();

```

- Convert the definition of hasStore from a boolean to a function.
- Update the test in `/test/specs/lib.environment.spec.js` and any other place where we call `hasStore` to call it as a function.

### Contributing to Mission Control

- [ ] Fork the repository from the [Mission Control GitHub page](https://github.com/DAVFoundation/missioncontrol).
- [ ] Clone a copy to your local machine with `$ git clone git@github.com:YOUR-GITHUB-USER-NAME/missioncontrol.git`
- [ ] Make sure you have _node.js_ and _npm_ installed on your machine. You can use this [guide](https://docs.npmjs.com/getting-started/installing-node) for help.
- [ ] Install all of the project's dependencies with npm. `$ cd missioncontrol; npm install`
- [ ] Run `npm test` to run linting checks and all the automated tests and see that they pass.
- [ ] **Code! code! code!**
- [ ] Before committing your code, run `npm test` one last time and make sure no errors (including linting errors) are thrown.
- [ ] Once you've made sure all your changes work correctly and committed all your changes, push your local changes back to github with `$ git push -u origin master`
- [ ] Visit your fork on GitHub.com ([https://github.com/YOUR-USER-NAME/missioncontrol](https://github.com/YOUR-USER-NAME/missioncontrol)) and create a pull request for your changes.
- [ ] Make sure your pull request describes exactly what you changed and references this issue (include the issue number in the title like this: `#28`)

### PR DESCRIPTION
**Pull request by Rob-Rychs**
Should close #70 

Gotta love that beautiful all green screen on the first try 😃 🚀 

screen shot 2017-09-22 at 2 59 33 am

### CODE REVIEW

File: server/lib/environment.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 const davEnv = process.env.DAV_ENV;
 
-const hasStore = ['simulated'].includes(davEnv);
+function hasStore() {
```
Comment (TalAter) [inline server/lib/environment.js]: I feel like a function expression would work more predictably than a function declaration... especially if we make any changes to the function in the future (which we will to support other environments).

You can read more about the difference here: https://kangax.github.io/nfe/ (I had to read it to remember the exact difference and nomenclature.)

In addition, with a function expression we can do an arrow functions, and those are simply 🔥 hot.

I'm thinking something like:
```javascript
const hasStore = () => ['simulated'].includes(davEnv);
```

**Comment by TalAter** [comment]
Small suggestion.

And yes... how I love watching that green tests passed message. I'm not kidding, it's such a good feeling. Part of the reason I love doing test driven development.

File: server/lib/environment.js (hunk 1 above)
Comment (Rob-Rychs) [reply server/lib/environment.js]: Wow. I had to read that article three times with a ten minute break in between each time for it to really sink in. Awesome bookmark for future reference though, thanks for sharing! 👍

**Comment by TalAter** [comment]
Sweet!

I love the feeling of moving from a solution that is a few lines long, and ending up with something H-O-T like that! 👍

**Pull request merged**